TYPO3.Media/Classes/TYPO3/Media/Domain/Service/ThumbnailService.php

```diff
@@ -82,62 +82,28 @@ public function getThumbnail(AssetInterface $asset, ThumbnailConfiguration $conf
     {
         $thumbnail = $this->thumbnailRepository->findOneByAssetAndThumbnailConfiguration($asset, $configuration);
```

Comment:
This means for a transient thumbnail we will always do a unecessary DB query. I don't have a quick fix in mind, just saying this should be improved performance wise...